osu.Game.Rulesets.Osu/Skinning/Legacy/OsuLegacySkinTransformer.cs

```diff
@@ -49,13 +49,16 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                         return followCircle;
 
                     case OsuSkinComponents.SliderBall:
-                        var sliderBallContent = this.GetAnimation("sliderb", true, true, animationSeparator: "");
+                        // specular and nd layers must come from the same source as the ball texure.
+                        var ballProvider = Source.FindProvider(s => s.GetTexture("sliderb") != null || s.GetTexture("sliderb0") != null);
+
+                        var sliderBallContent = ballProvider.GetAnimation("sliderb", true, true, animationSeparator: "");
 
                         // todo: slider ball has a custom frame delay based on velocity
                         // Math.Max((150 / Velocity) * GameBase.SIXTY_FRAME_TIME, GameBase.SIXTY_FRAME_TIME);
 
                         if (sliderBallContent != null)
-                            return new LegacySliderBall(sliderBallContent);
+                            return new LegacySliderBall(sliderBallContent, ballProvider);
```

Comment:
Note that doing this means `sliderb-nd` and `sliderb-spec` cannot be overriden by a user skin while still using the classic skin's slider ball animation, while osu!stable supports that iirc. Though I doubt there is any skin that does that nowadays, worth mentioning still.

Reply:
Sounds fine. If someone relies on this they can just copy the sliderb files from the template skin